core/Pos/Core/Block/Union/Instances.hs

```diff
@@ -58,4 +65,40 @@ instance HasDifficulty Block where
 -- IsHeader
 ----------------------------------------------------------------------------
 
-instance Bi BlockHeader  => IsHeader BlockHeader
+instance Bi BlockHeader => IsHeader BlockHeader
+
+----------------------------------------------------------------------------
+-- HasPrevBlock
+----------------------------------------------------------------------------
+
+instance HasPrevBlock (ComponentBlock a) where
+    prevBlockL = lens getter setter
+        where
+            getter (ComponentBlockGenesis genesisHeader) = genesisHeader ^. prevBlockL
+            getter (ComponentBlockMain mainHeader _)     = mainHeader ^. prevBlockL
+            setter (ComponentBlockGenesis genesisHeader) e =
+                ComponentBlockGenesis (genesisHeader & prevBlockL .~ e)
+            setter (ComponentBlockMain mainHeader payload) e =
+                ComponentBlockMain (mainHeader & prevBlockL .~ e) payload
```

Comment:
Hm, we use a bit different indentation for `where` clauses: https://github.com/serokell/serokell-util/blob/master/serokell-style.md#indentation